File: WatchBox/Shared/Models/MediaType.swift
//
//  MediaType.swift
//  SceneBox
//
//  Created by SpontaneousArray on 31.07.26.
//

import Foundation

nonisolated public enum MediaType: String, Sendable, Codable {
    case movie, series, anime

    public var browseTitle: String {
        switch self {
        case .movie: "Movies"
        case .series: "TV Shows"
        case .anime: "Anime"
        }
    }
}

File: lib/core/models/media_type.dart
enum MediaType {
  movie,
  series,
  anime;

  String get value => switch (this) {
        MediaType.movie => 'movie',
        MediaType.series => 'series',
        MediaType.anime => 'anime',
      };

  String get browseTitle => switch (this) {
        MediaType.movie => 'Movies',
        MediaType.series => 'TV Shows',
        MediaType.anime => 'Anime',
      };

  static MediaType fromValue(String? value) => switch (value) {
        'series' => MediaType.series,
        'anime' => MediaType.anime,
        _ => MediaType.movie,
      };
}
